backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java

```diff
@@ -2532,24 +2560,41 @@ public ImportBomRequestPreparation prepareImportBom(User user, String attachment
 
                 ImportBomRequestPreparation importBomRequestPreparation = spdxBOMImporter.prepareImportSpdxBOMAsRelease(sourceFile);
                 if (RequestStatus.SUCCESS.equals(importBomRequestPreparation.getRequestStatus())) {
-                    String name = importBomRequestPreparation.getName();
-                    String version = importBomRequestPreparation.getVersion();
-                    if (!isDuplicate(name, true)) {
-                        importBomRequestPreparation.setIsComponentDuplicate(false);
-                        importBomRequestPreparation.setIsReleaseDuplicate(false);
-                    } else if (!isDuplicate(name, version)) {
-                        importBomRequestPreparation.setIsComponentDuplicate(true);
-                        importBomRequestPreparation.setIsReleaseDuplicate(false);
-                    } else {
+                    List<String> componentsName = getComponentsName(importBomRequestPreparation.getComponentsName());
+                    Map<String, String> releasesName = getReleasesName(importBomRequestPreparation.getReleasesName());
+                    isDuplicateRelease(releasesName);
+                    isDuplicateComponent(componentsName,true);
+                    if (listComponentName.size() == 0 && mapReleaseName.size() == 0){
                         importBomRequestPreparation.setIsComponentDuplicate(true);
                         importBomRequestPreparation.setIsReleaseDuplicate(true);
                     }
+                    else {
+                        String componentName = "";
+                        String releaseName = "";
+                        if (listComponentName.isEmpty()){
+                            componentName = "Don't have Component created!";
```

Comment:
please extract a constant of the hardcoded text. Same in line 1580

Reply:
Hi @smrutis1 

I have pushed the discussed changes. Please check.